packages/@o3r/eslint-plugin/src/rules/o3r-widget-tags/o3r-widget-tags.ts

```diff
@@ -0,0 +1,303 @@
+      suggestParamMissing: 'Add @o3rWidgetParam {{ o3rWidgetParam }}.',
+      suggestRemoveDuplicatedO3rWidget: 'Remove the 2nd @o3rWidget.',
+      suggestRemoveDuplicatedO3rWidgetParam: 'Remove the 2nd @o3rWidgetParam.',
+      suggestAddO3rWidgetTag: 'Add @o3rWidget tag.',
+      suggestReplaceO3rWidgetType: 'Replace {{ currentType }} by {{ suggestedType }}.'
+    }
+  },
+  defaultOptions: [],
+  create: (context) => {
+    const options: O3rWidgetTagsRuleOption = context.options.reduce((acc, option) => ({ ...acc, ...option }), {});
+    const supportedO3rWidgets = new Set(Object.keys(options));
+    return {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      TSPropertySignature: (node) => {
+        const sourceCode = context.getSourceCode();
+        const [comment] = sourceCode.getCommentsBefore(node);
+
+        if (!comment || !comment.value.length) {
+          return;
+        }
+
+        const { loc, value: docText } = comment;
+        const widgetTypes = Array.from(docText.matchAll(/@o3rWidget (.*)/g));
+
+        if (widgetTypes.length > 1) {
+          const fix: TSESLint.ReportFixFunction = (fixer) => {
+            return fixer.replaceTextRange(comment.range, createCommentString(comment.value.replace(/(.*(@o3rWidget ).*(\n.*)*)(\n.*)\2.*/, '$1')));
+          };
+          return context.report({
+            messageId: 'onlyOneWidgetAllowed',
+            node,
+            loc,
+            fix,
+            suggest: [{
+              messageId: 'suggestRemoveDuplicatedO3rWidget',
+              fix
+            }]
+          });
+        }
+        const widgetType = widgetTypes[0]?.[1].trim();
+        const widgetParameterTexts = Array.from(docText.matchAll(/@o3rWidgetParam (.*)/g))
+          .map((match) => match[1].trim());
+
+        if (!widgetType) {
+          if (widgetParameterTexts.length) {
+            const fix: TSESLint.ReportFixFunction = (fixer) => {
+              return fixer.replaceTextRange(
+                comment.range,
+                createCommentString(comment.value.replace(/((.*)@o3rWidgetParam .*)/, '$2@o3rWidget widgetType\n$1'))
+              );
+            };
+            return context.report({
+              messageId: 'noParamWithoutWidget',
+              node,
+              loc,
+              fix,
+              suggest: [{
+                messageId: 'suggestAddO3rWidgetTag',
+                fix
+              }]
+            });
+          }
+          return;
+        }
+
+        const interfaceDeclNode = node.parent?.parent;
+        if (
+          interfaceDeclNode?.type !== TSESTree.AST_NODE_TYPES.TSInterfaceDeclaration
+          || !interfaceDeclNode.extends?.some((ext) =>
+            ext.type === TSESTree.AST_NODE_TYPES.TSInterfaceHeritage
+            && ext.expression.type === TSESTree.AST_NODE_TYPES.Identifier
+            && ['Configuration', 'NestedConfiguration', 'DynamicConfiguration'].includes(ext.expression.name)
```

Comment:
Not sure `DynamicConfiguration` exists.
Maybe `['Configuration', 'NestedConfiguration', 'DynamicConfiguration']` should be set as default value of a config to be overridable